[MonologBridge] Uninstallable together with symfony/http-kernel in 3.0.6

nex-dabe opened the pull request:
| Q | A |
| --- | --- |
| Branch? | 3.0 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Caused by #18705, it is impossible to install v3.0.6 of symfony/monolog-bridge
together with v3.0.6 of symfony/http-kernel.

The intention of #18705 "added a conflict between Monolog bridge 2.8 and
HTTP Kernel 3.0+" was to prevent installing symfony/monolog-bridge from the
3.0 series with http-kernel from the 2.8 series of symfony. While this now
works correctly in v2.8.6, it breaks installing symfony/monolog-bridge v3.0.6
with symfony/http-kernel v3.0.6.

This PR resolves this issue.
# How to reproduce
- Create a test directory and change into it - e.g. with `mkdir /tmp/reproduce-symfony-18745 && cd /tmp/reproduce-symfony-18745`
- Add the following composer.json to this test directory

```
{
   "require": {
     "symfony/monolog-bridge": "3.0.6",
     "symfony/http-kernel": "3.0.6"
  }
}
```
- Run `composer install` from the test directory
## Expected behavior

Composer installs symfony/monolog-bridge and symfony/http-kernel (together with their dependencies).
## Actual behavior

Composer fails with the following error messages:

```
#:/tmp/reproduce-symfony-18745$ composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for symfony/monolog-bridge 3.0.6 -> satisfiable by symfony/monolog-bridge[v3.0.6].
    - symfony/http-kernel v3.0.6 conflicts with symfony/monolog-bridge[v3.0.6].
    - Installation request for symfony/http-kernel 3.0.6 -> satisfiable by symfony/http-kernel[v3.0.6].
```

ymc-dabe commented:
Note that I'm actually not sure if it is right to prevent installing symfony/monolog-bridge v3.0.6 together with symfony/http-kernel v2.8.6, since PR #18705 only mentions a problem the other way around (monolog-bridge v2.8.x conflicts with http-kernel v3.0+).
If it is okay to install a 3.0.x monolog-bridge with an older 2.8.x http-kernel, the conflict in composer.json should be removed at all in branch 3.0. In that case, please let me know, so I can update this PR accordingly.

stof commented:
yeah, the conflict rule should be removed entirely in the 3.0 branch, which was forgotten when merging 2.8 into 3.0

Commit:
[MonologBridge] Uninstallable together with symfony/http-kernel in 3.0.6

Caused by #18705, it is impossible to install v3.0.6 of symfony/monolog-bridge
together with v3.0.6 of symfony/http-kernel.

The intention of #18705 "added a conflict between Monolog bridge 2.8 and
HTTP Kernel 3.0+" was to prevent installing symfony/monolog-bridge from the
3.0 series with http-kernel from the 2.8 series of symfony. While this now
works correctly in v2.8.6, it breaks installing symfony/monolog-bridge v3.0.6
with symfony/http-kernel v3.0.6.

This commit resolves this issue.

ymc-dabe commented:
Thanks for the feedback @stof. I updated this PR to remove the conflict with symfony/http-kernel in src/Symfony/Bridge/Monolog/composer.json.

Tobion commented:
:+1: 
Status: Reviewed

xabbuh commented:
👍

fabpot commented:
Thank you @ymc-dabe.

derrabus commented:
So, the workaround for Silex projects (and other projects using the `symfony/monolog-bridge` package directly) would be to mark the 3.0.6 release of that package as conflicting. There are no other changes between 3.0.5 and 3.0.6 anyway and composer will otherwise downgrade HttpKernel to 2.8.

```json
"conflict": {
    "symfony/monolog-bridge": "3.0.6"
}
```
